provisioning/miniprov/test/test_validator.py

```diff
@@ -0,0 +1,163 @@
+      "storage_set_id": 0
+    }
+  },
+  "cluster": {
+    "e766bd52-c19c-45b6-9c91-663fd8203c2e": {
+      "name": "cortx-cluster",
+      "storage_set": [
+          { "name": "StorageSet-1",
+           "server_nodes": [ "srvmachine-1" ]
+          }
+       ]
+    },
+    "storage_sets": {
+      "storage-set-1": [
+        "srvnode-1"
+      ]
+    },
+    "server_nodes": {
+      "7c4fd75dfedd7662e6a39b0a53274922": "srvnode-1"
+    },
+    "cluster_id": "e766bd52-c19c-45b6-9c91-663fd8203c2e",
+    "srvnode-1": {
+      "storage_set_id": 0,
+      "hostname": "ssc-vm-1623.colo.seagate.com",
+      "node_type": "VM",
+      "roles": [
+        "primary",
+        "openldap_server",
+        "kafka_server"
+      ],
+      "is_primary": "true",
+      "bmc": {
+      },
+      "network": {
+        "mgmt": {
+          "interfaces": [
+            "eth0"
+          ],
+        },
+        "data": {
+          "public_interfaces": [
+            "eth0"
+          ],
+          "private_interfaces": [
+            "eth0"
+          ],
+          "interface_type": "tcp",
+          "transport_type": "lnet",
+          "private_ip": "",
+          "roaming_ip": "127.0.0.1"
+        }
+      },
+      "storage": {
+        "enclosure_id": "enclosure-1",
+        "metadata_devices": [
+          "/dev/sdb"
+        ],
+        "data_devices": [
+          "/dev/sdd",
+          "/dev/sde",
+          "/dev/sdf",
+          "/dev/sdg"
+        ]
+      },
+      "s3_instances": 1
+    }
+  }}
+
+
+def update_machine(machine_id: str, hostname: str):
+    with open ("/tmp/test-conf-store.json", "w+") as conf_store:
+        json.dump(conf_store_data, conf_store)
+
+    with open("/tmp/test-conf-store.json", "r+") as jf:
+        data = json.load(jf)
+        new_machine = {"server_node": {
+                                       f'{machine_id}': {
+                                         "name": "srvnode-1",
+                                         "hostname": f'{hostname}',
+                                         "cluster_id": "e766bd52-c19c-45b6-9c91-663fd8203c2e",
+                                         "storage_set_id": 0
+                                       }
+                                      }
+                      }
+        data.update(new_machine)
+        data['cluster']['e766bd52-c19c-45b6-9c91-663fd8203c2e']['storage_set'][0]['server_nodes'] = [f'{machine_id}']
+        with open("/tmp/temp-test-conf-store.json", "w+") as ujf:
+            json.dump(data, ujf)
+
+class TestValidator(unittest.TestCase):
+    def test_is_cluster_first_node(self):
+        conf = ConfStoreProvider(URL)
+        hostname = socket.gethostname()
+        machine_id = conf.get_machine_id()
+        update_machine(machine_id, hostname)
+        validator = Validator(
+                        ConfStoreProvider("json:///tmp/temp-test-conf-store.json"))
+        assert validator.is_first_node_in_cluster()
```

Comment:
[`self.assertEqual()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual) please

`assert` instruction can be thrown away by the compiler. Anyway, I would suggest to use the framework-related API calls. `unittest` is our API for tests. Let's use it.

Reply:
Agree, thanks. I missed this.